dowhy/gcm/independence_test/ablation.py

```diff
@@ -0,0 +1,107 @@
+    are types of dependencies that this test can not detect. For instance, if X determines the variance of Y, then this
+    can not be captured. For these more complex dependencies, consider using the kernel_based independence test instead.
+
+    This test is motivated by Granger causality, the approx_kernel_based test and the following paper:
+        K Chalupka, P Perona, F. Eberhardt. *Fast Conditional Independence Test for Vector Variables with Large Sample Sizes*. arXiv:1804.02747, 2018.
+
+    :param X: Input data for X.
+    :param Y: Input data for Y.
+    :param Z: Input data for Z.
+    :param num_components_all_inputs: Number of kernel features when combining X and Z. Half of the number is used when
+                                      compared with Z alone. If Z is not given, Z will be ignored.
+    :param num_runs: Number of runs. This equals the number of estimated p-values, which get adjusted by the
+                     p_value_adjust_func.
+    :param p_value_adjust_func: A callable that expects a numpy array of multiple p-values and returns one p-value. This
+                                is typically used a family wise error rate control method.
+    :param f_test_samples_ratio: Ratio for splitting the data into test and training data sets for calculating
+                                 the f-statistic. A ratio of 0.3 means that 30% of the samples are used for the f-test (test samples) and 70% are
+                                 used for training the prediction model (training samples). If set to None, training and test data set are the same,
+                                 which could help in settings where only a few samples are available.
+    :param max_samples_per_run: Maximum number of samples used per run.
+    :return: The p-value for the null hypothesis that X and Y are independent given Z. If not Z is given,
+             then for the hypothesis that X and Y are independent.
+    """
+    if num_components_all_inputs < 2:
+        raise ValueError("Need at least two components for all inputs, but only %d were given!"
+                         % num_components_all_inputs)
+
+    X, Y = shape_into_2d(X, Y)
+
+    X = scale(apply_one_hot_encoding(X, fit_one_hot_encoders(X)))
+    Y = scale(apply_one_hot_encoding(Y, fit_one_hot_encoders(Y)))
+
+    if Z is not None:
+        Z = shape_into_2d(Z)
+        Z = scale(apply_one_hot_encoding(Z, fit_one_hot_encoders(Z)))
+
+    org_X = X
+    org_Y = Y
+    org_Z = Z
+
+    all_p_values = []
+    for _ in range(num_runs):
+        if X.shape[0] > max_samples_per_run:
+            random_indices = np.random.choice(X.shape[0], max_samples_per_run, replace=False)
+            X = org_X[random_indices]
+            Y = org_Y[random_indices]
+            if org_Z is not None:
+                Z = org_Z[random_indices]
+
+        if Z is not None:
+            all_inputs = Nystroem(n_components=num_components_all_inputs).fit_transform(np.column_stack([X, Z]))
```

Comment:
any justification why Nystroem used and not the other kernels in sklearn?

I'm also thinking from user flexibility perspective. Would it make sense to expose "kernel_approx_func" as an argument of this function, so users can input whatever kernel they prefer, while Nystroem remains the default?